src/Razor/src/Microsoft.AspNetCore.Razor.Language/Components/ComponentBindLoweringPass.cs

```diff
@@ -53,16 +55,63 @@ protected override void ExecuteCore(RazorCodeDocument codeDocument, DocumentInte
                     continue;
                 }
 
-                if (node.TagHelper.IsBindTagHelper() && node.AttributeName.StartsWith("bind"))
+                if (node.TagHelper.IsBindTagHelper() && node.AttributeName.StartsWith("bind") && !node.IsParameterMatch)
                 {
-                    // Workaround for https://github.com/aspnet/Blazor/issues/703
-                    var rewritten = RewriteUsage(reference.Parent, node);
-                    reference.Remove();
+                    bindEntries[node.AttributeName] = new BindEntry(reference);
+                }
+            }
+
```

Comment:
So we're doing a lot of work in the below sections of code to find the base `bind` attributes and then locate the parameter `bind` attributes and validate their usage/extract information. Is that pattern something we expect of future directive attributes? Basically if it's a common pattern where parameters are usually only valuable if there's a base directive attribute to associate we may want to consider baking it into Razor vs. being so special cased for bind here (we'd still need to be special cased but not as much)

Reply:
This might be overkill at this point. We can always do this later when this is actually necessary.

Reply:
This will be a common pattern.